openassessment/workflow/models.py

```diff
@@ -316,17 +322,23 @@ def update_from_assessments(self, assessment_requirements):
 
         new_staff_score = self.get_score(assessment_requirements, {'staff': step_for_name.get('staff', None)})
         if new_staff_score:
-            old_score = self.score
+            old_score = sub_api.get_latest_score_for_submission(self.submission_uuid)
             if not old_score or old_score['points_earned'] != new_staff_score['points_earned']:
```

Comment:
Is it possible that old_score['points_earned'] == new_staff_score['points_earned'], but old_score was not a staff score? If so, don't we still want to go through this block even though the numerical grade is the same?

(I wasn't part of the original review, so sorry if I'm completely confused here).